core/templates/dev/head/pages/creator_dashboard/CreatorDashboard.js

```diff
@@ -167,6 +172,138 @@ oppia.controller('CreatorDashboard', [
       return value;
     };
 
+    var _fetchMessages = function(threadId) {
+      $http.get('/threadhandler/' + threadId).then(function(response) {
+        var allThreads = $scope.mySuggestionsList.concat(
+          $scope.suggestionsToReviewList);
+        for (var i = 0; i < allThreads.length; i++) {
+          if (allThreads[i].threadId === threadId) {
+            allThreads[i].setMessages(response.data.messages);
+            break;
+          }
+        }
+      });
+    };
+
+    $scope.clearActiveThread = function() {
+      $scope.activeThread = null;
+    };
+
+    $scope.setActiveThread = function(threadId) {
+      _fetchMessages(threadId);
+      for (var i = 0; i < $scope.mySuggestionsList.length; i++) {
+        if ($scope.mySuggestionsList[i].threadId === threadId) {
```

Comment:
mySuggestionsList contains suggestions made by the user and suggestionsToReviewList contains the suggestions they can review, right? If the thread is for a suggestion that we have made, do we need to even check suggestionsToReviewList?

Reply:
You are right about the first point. I don't think I understood what you meant by the second sentence. You mean, in these loops to setActiveThread, if we should check suggestionsToReviewList or not?

Reply:
Done